pygeodiff/geodifflib.py

```diff
@@ -315,6 +352,51 @@ def create_changeset_dr(self, driver_src, driver_src_info, src, driver_dst, driv
         res = func(b_string1, b_string2, b_string3, b_string4, b_string5, b_string6, b_string7)
         _parse_return_code( res, "CreateChangesetDr" )
 
+    def apply_changeset_ex(self, driver, driver_info, base, changeset):
+        res = self.lib.GEODIFF_applyChangesetEx(
+            ctypes.c_char_p(driver.encode('utf-8')),
+            ctypes.c_char_p(driver_info.encode('utf-8')),
+            ctypes.c_char_p(base.encode('utf-8')),
+            ctypes.c_char_p(changeset.encode('utf-8')))
+        _parse_return_code(res, "apply_changeset_ex")
+
+    def create_rebased_changeset_ex(self, driver, driver_info, base, base2modified, base2their, rebased, conflict_file):
+        res = self.lib.GEODIFF_createRebasedChangesetEx(
+            ctypes.c_char_p(driver.encode('utf-8')),
+            ctypes.c_char_p(driver_info.encode('utf-8')),
+            ctypes.c_char_p(base.encode('utf-8')),
+            ctypes.c_char_p(base2modified.encode('utf-8')),
+            ctypes.c_char_p(base2their.encode('utf-8')),
+            ctypes.c_char_p(rebased.encode('utf-8')),
+            ctypes.c_char_p(conflict_file.encode('utf-8')))
+        _parse_return_code(res, "create_rebased_changeset_ex")
+
+    def rebase_ex(self, driver, driver_info, base, modified, base2their, conflict_file):
+        res = self.lib.GEODIFF_rebaseEx(
+            ctypes.c_char_p(driver.encode('utf-8')),
+            ctypes.c_char_p(driver_info.encode('utf-8')),
+            ctypes.c_char_p(base.encode('utf-8')),
+            ctypes.c_char_p(modified.encode('utf-8')),
+            ctypes.c_char_p(base2their.encode('utf-8')),
+            ctypes.c_char_p(conflict_file.encode('utf-8')))
+        _parse_return_code(res, "rebase_ex")
+
+    def dump_data(self, driver, driver_info, src, changeset):
+        res = self.lib.GEODIFF_dumpData(
+            ctypes.c_char_p(driver.encode('utf-8')),
+            ctypes.c_char_p(driver_info.encode('utf-8')),
+            ctypes.c_char_p(src.encode('utf-8')),
+            ctypes.c_char_p(changeset.encode('utf-8')))
+        _parse_return_code(res, "dump_data")
+
+    def schema(self, driver, driver_info, src, json):
```

Comment:
driver_info can be nullprt in the c++ api?

Reply:
yes, in all api calls the driver info can be null (because at least for sqlite there's no need for it)